### PR TITLE
Use free GitHub Actions workers whenever possible

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
This will slow down tests considerably; let's see if it's acceptable.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -57,16 +57,16 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest-8-cores]
+        os: [ubuntu-latest]
```
Comment (0x2b3bfa0) [inline .github/workflows/tests.yml:60]: Let's leave this one for a faster feedback loop?

```suggestion
        os: [ubuntu-latest-8-cores]
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -207,7 +207,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest-16-cores, macos-latest, windows-latest-8-cores]
+        os: [ubuntu-latest, macos-latest, windows-latest]
```
Comment (mattseddon) [inline .github/workflows/tests.yml:210]: I'm tipping that `ubuntu-latest` is running out of memory for the multimodal example test

**Review by mattseddon — APPROVED** [review]
I'm fine with this as long as everything passes

**Review by dtulga — APPROVED** [review]
LGTM, as long as all the tests pass.

**Comment by shcheklein** [comment]
@0x2b3bfa0 what is the progress on this? can someone pick it up? is it good to be merged?

**Review by dreadatour — APPROVED** [review]
Sounds like a good plan! 👍

**Pull request merged**